classes/task/sendfailednotifications_task.php

```diff
@@ -0,0 +1,173 @@
+        $emailmsg = "";
+
+        // styles
+        $emailmsg .= '<style type="text/css">.header {text-align:left;}</style>';
+    
+        // first line
+        $emailmsg .= \html_writer::tag('p', get_string('failedtosend', 'logstore_xapi'));
+    
+        // summary info
+        $endpointname = get_string('endpoint', 'logstore_xapi');
+        $url = get_config('logstore_xapi', 'endpoint');
+        $endpointurl = \html_writer::tag('a', $url, array('target' => '_blank', 'href' => $url));
+    
+        $errorlogpage = get_string('errorlogpage', 'logstore_xapi');
+        $url = new \moodle_url("/admin/tool/log/store/xapi/report.php");
+        $errorlogurl = \html_writer::tag('a', $url, array('target' => '_blank', 'href' => $url));
+    
+        // first table
+        $table = new \html_table();
+    
+        // data
+        $table->data[] = array($endpointname, $endpointurl);
+        $table->data[] = array($errorlogpage, $errorlogurl);
+    
+        // add table to message
+        $emailmsg .= \html_writer::table($table);
+    
+        // separator
+        $emailmsg .= \html_writer::tag('h3', get_string('failurelog', 'logstore_xapi'));
+    
+        // second table
+        $table = new \html_table();
+
+        // header
+        $heading1 = get_string('datetimegmt', 'logstore_xapi');
+        $heading2 = get_string('eventname', 'logstore_xapi');
+        $heading3 = get_string('response', 'logstore_xapi');
+        $table->head = array($heading1, $heading2, $heading3);
+    
+        // data
+        foreach ($results as $result) {
+            $col1 = userdate($result->timecreated);
+            $col2 = $result->eventname;
+            $col3 = $result->response;
+            $table->data[] = array($col1, $col2, $col3);
+        }
+    
+        // add table to message
+        $emailmsg .= \html_writer::table($table);
+    
+        return $emailmsg;
+    }
+
+    /**
+     * Send email using email_to_user.
+     *
+     * @return int 1 = sent, 0 = not sent
+     */
+    private function sendmail($msg, $subject, $emailto) {
+
+        $user = new \stdClass();
+        $user->id = self::DEFAULT_RECEIVER;
+        $user->username = self::DEFAULT_RECEIVER_NAME;
+        $user->email = $emailto;
+        $user->deleted = 0;
+        $user->mailformat = FORMAT_HTML;
+    
+        $from = new \stdClass();
+        $from->id = self::DEFAULT_SENDER;
+        $from->username = self::DEFAULT_SENDER_NAME;
+        $from->email = self::DEFAULT_SENDER_EMAIL;
+        $from->deleted = 0;
+        $from->mailformat = FORMAT_HTML;
+    
+        $messageid = email_to_user($user, $from, $subject, html_to_text($msg), $msg);
+        return $messageid;
+    }
+
+    /**
+     * Do the job.
+     * Throw exceptions on errors (the job will be retried).
+     */
+    public function execute() {
+        $manager = get_log_manager();
+        $store = new store($manager);
+
+        echo "In send failed notifications task execute".PHP_EOL;
```

Comment:
Do we really need this anymore? Can it be removed, or conditional based on a debug flag?